Content.Shared/Pulling/Systems/SharedPullingStateManagementSystem.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Content.Shared.GameTicking;
+using Content.Shared.Input;
+using Content.Shared.Physics.Pull;
+using Content.Shared.Pulling.Components;
+using Content.Shared.Pulling.Events;
+using JetBrains.Annotations;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Input.Binding;
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Dynamics.Joints;
+using Robust.Shared.Players;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Pulling
+{
+    /// <summary>
+    /// This is the core of pulling state management.
+    /// Because pulling state is such a mess to get right, all writes to pulling state must go through this class.
+    /// </summary>
+    [UsedImplicitly]
+    public class SharedPullingStateManagementSystem : EntitySystem
+    {
+        // A WARNING:
+        // The following 2 functions are the most internal part of the pulling system's relationship management.
+        // They do not expect to be cancellable.
+        private void ForceDisconnect(SharedPullerComponent puller, SharedPullableComponent pullable)
+        {
+            var eventBus = pullable.Owner.EntityManager.EventBus;
+
+            var pullerPhysics = puller.Owner.GetComponent<PhysicsComponent>();
+            var pullablePhysics = pullable.Owner.GetComponent<PhysicsComponent>();
+
+            // MovingTo shutdown
+            ForceSetMovingTo(pullable, null);
+
+            // Joint shutdown
+            if (pullerPhysics.Joints.Contains(pullable.PullJoint!))
+            {
+                pullerPhysics.RemoveJoint(pullable.PullJoint!);
+            }
+            pullable.PullJoint = null;
+
+            // State shutdown
+            puller.Pulling = null;
+            pullable.Puller = null;
+
+            // Messaging
+            var message = new PullStoppedMessage(pullerPhysics, pullablePhysics);
+
+            eventBus.RaiseLocalEvent(puller.Owner.Uid, message, broadcast: false);
+            eventBus.RaiseLocalEvent(pullable.Owner.Uid, message);
```

Comment:
Just use the EntitySystem `RaiseLocalEvent` methods.

Reply:
yeaaaah, that kinda got inherited during the refactor, fixed now